### PR TITLE
Delivers capstone project

### PR DESCRIPTION
**Pull request by King-Benx**
**Delivers the following:**
1. API integration
2. Popup
3. Responsiveness
4. walkthrough
5. Linting
6. webpack
7. Counter helper function for counting shows and comments respectively
8. Tests for counter
9. Update like and likes depending on number

**Live link**
[https://king-benx.github.io/your-movies-capstone-project/](https://king-benx.github.io/your-movies-capstone-project/l)

**How to test:**
1. Clone the project ```git clone git@github.com:King-Benx/your-movies-capstone-project.git```
11. Checkout to the root directory ```cd your-movies-capstone-project```
12. Checkout to the dev ```git checkout dev```
13. Install dependencies ```npm i```
14. Run the project using webpack ```npm run dev```

**Screenshots:**
1. Mobile
Screenshot 2023-03-17 at 7 51 47 AM

2. Desktop

Screenshot 2023-03-17 at 07-51-23 Meal Capstone

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    <div class="container">
+      <!-- Header -->
+      <header>
+        <a href="">
+          <img
+            src="https://placehold.co/40x40/orange/white?font=Lora&text=YAS"
+            alt=""
+          >
+          <span>Your Amazing Shows</span>
+        </a>
```
Comment (binhussen) [inline src/index.html:19]: - Kindly, you missed showing the number of movies :+1: so please add a count function that counts the number of movies and displays it, and doesn't use API response.length to show the movie count instead count the card list because you will use this function later for  testing :+1: 
Screenshot from 2023-03-18 20-54-32
Screenshot from 2023-03-17 22-40-07

File: src/index.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      <!-- Dialog -->
+      <dialog class="modal" id="modal">
+        <span class="modal-close">x</span>
+        <div class="card-image">
+          <img src="https://placehold.co/400x400/" id="card-image" alt="">
+        </div>
+        <div class="modal-body">
+          <h3 id="modal-name">name</h3>
+          <p id="modal-description"></p>
+          <h4>Comments <span id="comment-number"></span></h4>
```
Comment (binhussen) [inline src/index.html]: - Kindly, you missed here also showing the number of comments given for that specific movie, so please add a count function that counts the number of comments and displays it, and you will add tests for this function later:+1:

File: src/modules/render.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                ></g>
+                <g id="SVGRepo_iconCarrier">
+                  <g>
+                    <path
+                      d="M433.601,67.001c-24.7-24.7-57.4-38.2-92.3-38.2s-67.7,13.6-92.4,38.3l-12.9,12.9l-13.1-13.1 c-24.7-24.7-57.6-38.4-92.5-38.4c-34.8,0-67.6,13.6-92.2,38.2c-24.7,24.7-38.3,57.5-38.2,92.4c0,34.9,13.7,67.6,38.4,92.3 l187.8,187.8c2.6,2.6,6.1,4,9.5,4c3.4,0,6.9-1.3,9.5-3.9l188.2-187.5c24.7-24.7,38.3-57.5,38.3-92.4 C471.801,124.501,458.301,91.701,433.601,67.001z M414.401,232.701l-178.7,178l-178.3-178.3c-19.6-19.6-30.4-45.6-30.4-73.3 s10.7-53.7,30.3-73.2c19.5-19.5,45.5-30.3,73.1-30.3c27.7,0,53.8,10.8,73.4,30.4l22.6,22.6c5.3,5.3,13.8,5.3,19.1,0l22.4-22.4 c19.6-19.6,45.7-30.4,73.3-30.4c27.6,0,53.6,10.8,73.2,30.3c19.6,19.6,30.3,45.6,30.3,73.3 C444.801,187.101,434.001,213.101,414.401,232.701z"
+                    ></path>
+                  </g>
+                </g>
+              </svg>
+              <span id=${id} class="modal-likes">${likes} Likes</span>
```
Comment (binhussen) [inline src/modules/render.js]: - Kindly, I am seeing that your API is updating the number of likes and fetching also from the API correctly, but you are showing always 1 if it have likes :+1:

File: dist/index.html [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+              <textarea
+                name="comment"
+                id="comment"
+                required
+                placeholder="Your insights"
+              >
+              </textarea>
+            </div>
+            <input type="hidden" id="item-id">
+            <button type="submit" id="submit">Comment</button>
```
Comment (binhussen) [inline dist/index.html:59]: - Kindly, after adding the new comment your page is redirecting to the home page but it should update itself and the new comment should be added to the comments list :+1: so please update it as it is part of the requirement
Screenshot from 2023-03-18 21-55-59

**Review by binhussen — CHANGES_REQUESTED** [review]
Hi @King-Benx and @DilsherB , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- All linter checks passed ✔️
- Used wepack ✔️
- Used Git flow ✔️

## Required Changes ♻️
- [Required] Kindly you didn't test for your count function, please add it as it is part of the [requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/group-capstone/js_capstone.md)
Screenshot from 2023-03-18 21-18-47

- _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: tests/Counter.test.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    const shows = [{
+      item_id: 1,
+      name: 'show 1',
+      description: 'show 1 description',
+    }, {
+      item_id: 2,
+      name: 'show 2',
+      description: 'show 2 description',
+    }];
+
```
Comment (Ghiftee) [inline tests/Counter.test.js:24]: - [ ] Great job adding tests for the counter functions, however, kindly ensure that you have tests for all edge cases, for instance, a scenario in which there are 0 items on the page (counter function should return 0)

File: src/modules/render.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+                ></g>
+                <g id="SVGRepo_iconCarrier">
+                  <g>
+                    <path
+                      d="M433.601,67.001c-24.7-24.7-57.4-38.2-92.3-38.2s-67.7,13.6-92.4,38.3l-12.9,12.9l-13.1-13.1 c-24.7-24.7-57.6-38.4-92.5-38.4c-34.8,0-67.6,13.6-92.2,38.2c-24.7,24.7-38.3,57.5-38.2,92.4c0,34.9,13.7,67.6,38.4,92.3 l187.8,187.8c2.6,2.6,6.1,4,9.5,4c3.4,0,6.9-1.3,9.5-3.9l188.2-187.5c24.7-24.7,38.3-57.5,38.3-92.4 C471.801,124.501,458.301,91.701,433.601,67.001z M414.401,232.701l-178.7,178l-178.3-178.3c-19.6-19.6-30.4-45.6-30.4-73.3 s10.7-53.7,30.3-73.2c19.5-19.5,45.5-30.3,73.1-30.3c27.7,0,53.8,10.8,73.4,30.4l22.6,22.6c5.3,5.3,13.8,5.3,19.1,0l22.4-22.4 c19.6-19.6,45.7-30.4,73.3-30.4c27.6,0,53.6,10.8,73.2,30.3c19.6,19.6,30.3,45.6,30.3,73.3 C444.801,187.101,434.001,213.101,414.401,232.701z"
+                    ></path>
+                  </g>
+                </g>
+              </svg>
+              <span id=${id} class="modal-likes">${likes} ${likes === 1 ? 'Like' : 'Likes'}</span>
```
Comment (Ghiftee) [inline src/modules/render.js:40]: - [ ] As the previous reviewer stated, no matter how many times the like button is clicked, it remains as 1 like. Kindly fix this.

File: src/index.html [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    <div class="container">
+      <!-- Header -->
+      <header>
+        <a href="">
+          <img
+            src="https://placehold.co/40x40/orange/white?font=Lora&text=YAS"
+            alt=""
+          >
+          <span>Your Amazing Shows</span>
+        </a>
```
Comment (Ghiftee) [inline src/index.html:19]: - [ ] Great work adding the number of movies, however, the requirements states that your project should look similar to the given design so kindly ensure the number of movies are placed as shown below:

### Given design
Screenshot 2023-03-18 at 23 56 00 (2)

### Your design
Screenshot 2023-03-18 at 23 55 51 (2)

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hi there 👋🏽 , 

Great work on making the changes requested by a previous reviewer 👏 
To highlight 👍🏽 
✔️ Counter functions are added
✔️ Tests are added
✔️ Comments are updated properly
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by medaminedev66** [comment]
Hello @King-Benx and and @DilsherB 👋,

_This is an additional review as a result of the second opinion request._

Well done on addressing the changes suggested by the previous reviewers. Nonetheless, as this project is a collaborative effort, it is essential that all requirements are fulfilled for the project to receive approval.

Per the request of @Ghiftee, the like button functionality is still not implemented yet. It is important to note that when a user clicks on the Like button, the interaction should be recorded in the Involvement API, and the screen should be updated accordingly. Kindly work on that so that your project is complete.

About testing, you already created a scenario expecting 2 items on the page. You can add another scenario in which there are 0 items on the page.

## Optional suggestions

Regarding the design, aligning the number of items on the same line as the logo is feasible. However, it is not mandatory, and you may choose to implement it at your convenience.

Additionally, there was a change requested by @binhussen to display the number of comments for a specific movie. While this change was not requested by @Ghiftee, I consider it optional but I strongly recommend implementing it for an improved user experience. Please ensure that the number of comments is displayed promptly after the user adds a comment.

Keep rocking 🚀!
